lib/rack/request.rb

```diff
@@ -16,7 +16,18 @@ class << self
       attr_accessor :ip_filter
     end
 
-    self.ip_filter = lambda { |ip| /\A127\.0\.0\.1\Z|\A(10|172\.(1[6-9]|2[0-9]|30|31)|192\.168)\.|\A::1\Z|\Afd[0-9a-f]{2}:.+|\Alocalhost\Z|\Aunix\Z|\Aunix:/i.match?(ip) }
+    TRUSTED_PROXIES = Regexp.union(
+      /\A127\.(([1-9]?\d|[12]\d\d)\.){2}([1-9]?\d|[12]\d\d)\Z/i,     # localhost IPv4 range 127.x.x.x, per RFC-3330
```

Comment:
Can we explain the reasoning for using `([1-9]?\d|[12]\d\d)` as a valid IPv4 segment?  Is it just to prevent preceding `0`?  That you can handle more simply with `[1-9]\d\d`.  If it is only allow valid IPv4 segments, then it isn't strict enough, since `256` will match, and you'll need a more complicated regexp, such as `([1-9]?\d|1\d\d)|2[0-4]\d|25[0-5]`.

Whatever regexp is used for IPv4 segments, it should be assigned to a local variable, and interpolated into each regexp used in the union to reduce duplication.

Additionally, IPv4 regexps don't need the case insensitive regexp flag, since they don't have any alphabetic characters, only numeric characters.

Reply:
Thank you @jeremyevans this is all very valid feedback.

I think I have addressed all of your concerns, and I have added more negative test cases to confirm that the IPv4 octet regexp is properly validating.  Please give it another look at your convenience.